### ISSUE TITLE
[BUG]: `abp-pcap-xgb` model has 98.7% false positive rate on real-world LAN traffic and cannot distinguish DDoS from legitimate SCP transfers

### ISSUE THREAD

**Issue by h-network** [nv-morpheus/Morpheus#2327]
### Version

v25.06

### Which installation method(s) does this occur on?

Docker

### Describe the bug.

# Bug Report: Pre-trained abp-pcap-xgb model flags 98.7% of real-world LAN traffic as anomalous

---

## Description

The pre-trained `abp-pcap-xgb` XGBoost model shipped with Morpheus v25.06 (via `morpheus-tritonserver-models:25.06`) produces a **98.7% false positive rate** when run against real-world LAN traffic. The model flags virtually all traffic as anomalous, including mDNS, DHCP, PostgreSQL, HTTPS, and SCP file transfers.

More critically, the model **cannot distinguish between a SYN flood DDoS attack and a legitimate SCP file transfer** — producing statistically identical output for both.

## Environment

- Morpheus v25.06 (`nvcr.io/nvidia/morpheus/morpheus:25.06-runtime`)
- Triton Inference Server (`nvcr.io/nvidia/morpheus/morpheus-tritonserver-models:25.06`)
- Model: `abp-pcap-xgb` (default, no modifications)
- GPUs: 2x RTX 5090 (Morpheus preprocessing), RTX 4070 Ti + RTX 3090 (Triton)
- Driver: 570.211.01, CUDA: 12.8
- Network: 192.168.178.0/24 LAN

### 1. Start Triton with the pre-trained model
```bash
docker run --rm --gpus=all -p8000:8000 -p8001:8001 -p8002:8002 \
  nvcr.io/nvidia/morpheus/morpheus-tritonserver-models:25.06 \
  tritonserver --model-repository=/models/triton-model-repo \
  --exit-on-error=false --model-control-mode=explicit --load-model abp-pcap-xgb
```

### 2. Capture live LAN traffic with tshark and convert to Morpheus JSON format
```bash
sudo tshark -T ek \
  -e frame.time_epoch -e eth.src -e eth.dst \
  -e ip.src -e ip.dst -e ip.proto \
  -e tcp.srcport -e tcp.dstport -e tcp.flags -e tcp.len \
  -i eth0 -c 5000 | jq -c -f tshark_to_morpheus.jq > live_capture.jsonlines
```

### 3. Run the PCAP detection pipeline (exactly as documented in examples/abp_pcap_detection)
```bash
morpheus --log_level INFO \
  --plugin "examples/abp_pcap_detection/abp_pcap_preprocessing.py" \
  run --pipeline_batch_size 100000 --model_max_batch_size 100000 \
  pipeline-fil --model_fea_length 13 --label=probs \
  from-file --filename live_capture.jsonlines --filter_null False \
  deserialize pcap-preprocess \
  inf-triton --model_name "abp-pcap-xgb" --server_url "localhost:8000" \
  add-class --label=probs \
  serialize \
  to-file --filename output.jsonlines --overwrite
```

### 4. Count results
```bash
grep -c '"probs":true' output.jsonlines   # → ~98% of all packets
grep -c '"probs":false' output.jsonlines  # → ~2% of all packets
```

## Test Results

Ran three scenarios through the identical pipeline, capturing on the same LAN interface:

### Scenario A: SYN Flood DDoS Attack
```bash
# Attacks launched during capture
hping3 -a 5.79.3.1 -i u1000 -S 192.168.178.101 -c 10000  # spoofed source
hping3 -i u1000 -S 192.168.178.101 -c 10000                # real source
```

| Run | Packets | Flagged | Flag Rate |
|-----|---------|---------|-----------|
| 1 | 4,782 | 4,581 | 95.7% |
| 2 | 4,819 | 4,651 | 96.5% |
| 3 | 4,958 | 4,930 | 99.4% |
| 4 | 4,996 | 4,996 | 100.0% |
| 5 | 4,993 | 4,993 | 100.0% |
| **Total** | **24,548** | **24,151** | **98.3%** |

### Scenario B: Legitimate SCP Transfer (Outbound, 100MB)
```bash
scp h-oracle:/tmp/test_transfer.bin /tmp/test_transfer.bin
```

| Packets | Flagged | Flag Rate |
|---------|---------|-----------|
| 4,987 | 4,972 | **99.6%** |

### Scenario C: Legitimate SCP Transfer (Inbound)
```bash
scp file h-titan:/tmp/
```

| Packets | Flagged | Flag Rate |
|---------|---------|-----------|
| 4,991 | 4,991 | **100.0%** |

### NVIDIA Sample Data (Baseline)

| Packets | Flagged | Flag Rate |
|---------|---------|-----------|
| 537,241 | 13,921 | **2.6%** |

## The Problem

| Scenario | Threat Level | Flag Rate |
|----------|-------------|-----------|
| SYN Flood DDoS | **Malicious** | 98.3% |
| SCP File Transfer (outbound) | Legitimate | 99.6% |
| SCP File Transfer (inbound) | Legitimate | 100.0% |
| Normal LAN Traffic | Legitimate | 98.8% |
| NVIDIA Sample Data | Mixed | 2.6% |

**The model produces identical output for a DDoS attack and a file copy.** The legitimate SCP transfers actually scored higher anomaly rates than the actual attack. The only scenario where the model produces meaningful results is the sample dataset it was trained on.

Traffic types incorrectly flagged as anomalous:
- mDNS multicast (224.0.0.251)
- SSDP (224.0.1.140)
- DHCP broadcasts (255.255.255.255)
- PostgreSQL (port 5432)
- HTTPS (port 443)
- NFS (port 2049)
- SCP/SSH (port 22)

## Expected Behavior

A pre-trained model shipped as part of an official example should either:
1. Produce meaningful results on common real-world LAN traffic, or
2. Be clearly documented as only working on the specific dataset provided, with guidance on retraining for other environments

The current documentation presents `abp-pcap-xgb` as a general-purpose anomaly detection model without noting this limitation.

## Actual Behavior

The model flags 98.7% of all real-world traffic as anomalous and cannot distinguish a DDoS attack from a file transfer.

## Suggested Improvements

1. **Document the limitation** — The example README should state that the model only works on traffic matching its training data and is not a general-purpose detector
2. **Provide retraining guidance** — Include instructions for training the model on a user's own network baseline
3. **Include a diverse pre-trained model** — Or at least a model trained on more representative traffic patterns (not just one Kubernetes cluster)
4. **Add a warning** — When the model flags >90% of traffic, it should warn the user that the model may not be calibrated for the target environment

## Additional Context

Full test reports with LLM-assisted analysis, pipeline architecture, and benchmark data are available at: 
https://github.com/h-network/h-sniffer

Tested with 36,872 real-world packets across multiple scenarios.

### Minimum reproducible example

```shell
# 1. Start Triton with pre-trained model
  docker run --rm --gpus=all -p8000:8000 -p8001:8001 -p8002:8002 \
    nvcr.io/nvidia/morpheus/morpheus-tritonserver-models:25.06 \
    tritonserver --model-repository=/models/triton-model-repo \
    --exit-on-error=false --model-control-mode=explicit --load-model abp-pcap-xgb

  # 2. Capture 500 packets and convert to Morpheus format
  sudo tshark -T ek -e frame.time_epoch -e ip.src -e ip.dst -e ip.proto \
    -e tcp.srcport -e tcp.dstport -e tcp.flags -e tcp.len \
    -i eth0 -c 500 | jq -c -f tshark_to_morpheus.jq > capture.jsonlines

  # 3. Run the ABP pipeline
  morpheus --log_level INFO \
    --plugin "examples/abp_pcap_detection/abp_pcap_preprocessing.py" \
    run --pipeline_batch_size 100000 --model_max_batch_size 100000 \
    pipeline-fil --model_fea_length 13 --label=probs \
    from-file --filename capture.jsonlines --filter_null False \
    deserialize pcap-preprocess \
    inf-triton --model_name "abp-pcap-xgb" --server_url "localhost:8000" \
    add-class --label=probs serialize \
    to-file --filename output.jsonlines --overwrite

  # 4. Check results
  grep -c '"probs":true' output.jsonlines
  grep -c '"probs":false' output.jsonlines
```

### Relevant log output

<details><summary>Click here to see error details</summary><pre>

 [Paste the error here, it will be hidden by default]

</pre></details>

### Full env printout

<details><summary>Click here to see environment details</summary><pre>

 [Paste the results of print_env.sh here, it will be hidden by default]

</pre></details>

### Code of Conduct

- [x] I agree to follow Morpheus' Code of Conduct
- [x] I have searched the [open bugs](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report
